includes/shortcodes/class-wc-shortcode-checkout.php

```diff
@@ -236,8 +236,9 @@ private static function order_received( $order_id = 0 ) {
 		// Empty awaiting payment session.
 		unset( WC()->session->order_awaiting_payment );
 
-		// In case order is created from admin, but paid by the actual customer, store the ip address of the payer.
-		if ( $order ) {
+		// In case order is created from admin, but paid by the actual customer, store the ip address of the payer
+		// when they visit the payment confirmation page.
+		if ( $order && 'admin' === $order->get_created_via() ) {
```

Comment:
I am thinking that maybe we should create a method like is_created_via and use that instead, as doing things this way sets us up for failure in the future if we need to change something related to the created_via field. It is difficult to cover backward compatibility with hardcoded comparisons like this. I guess this is more related to the use of external code but we should be setting the example in core.

Reply:
I see your point and kind of agree with you, esp now that you have the perspective from statuses.
 
Still a bit on the fence about creating more abstractions for the future 'maybes', but going to go with your suggestion. Please see my refined approach in fa38299.

Reply:
I see your point and agree with what you're saying, especially now that you have fresh experience from order statuses.

I'm still a bit on the fence about introducing more abstractions for future 'maybes', though. Please see my update in fa38299.